wasi/src/Module.scala

```diff
@@ -0,0 +1,470 @@
+package swam
+package wasi
+
+import Types._
+import Header._
+import cats.Applicative
+import cats.effect._
+import swam.runtime.formats._
+import swam.runtime.formats.DefaultFormatters._
+import cats.effect.IO
+import swam.runtime.Memory
+import swam.runtime.imports.annotations.{effect, effectful, module, pure}
+
+@module
+abstract class Module[@effect F[_]](implicit F: Applicative[F]) {
+  var mem: Memory[IO] = null
+
+  val name = "wasi_snapshot_preview1"
+
+  def tryToExecute(a: => errnoEnum.Value) = {
+    try a.id
+    catch {
+      case x: WASIException => x.errno.id
+    }
+  }
```

Comment:
The all point of having the `F` is to not use try/catch, but to handle effects and exceptions in the context of `F`. This all file is to javascript like, and I would love to have a pure implementation of it. It is a lot, I will probably check it out and rewrite it as it involves some advanced cats-effect elements.